app/Http/Controllers/CampaignController.php

```diff
@@ -37,7 +39,20 @@ public function create(Request $request)
      */
     public function store(CampaignStoreRequest $request)
     {
-        $campaign = Campaign::create($request->validated());
+        $data = $request->validated();
+
+        $campaign = Campaign::create($data);
+
+        //user joins capaign
+        $user = Auth::user();
+
+        if ($request->has('character_id')) {
+            $user->campaigns()->attach($campaign, ['is_dm' => false]);
+            $character = Character::findOrFail($data['character_id'])->first();
```

Comment:
You can omit the `first()` here
```suggestion
            $character = Character::findOrFail($data['character_id']);
```